crates/runtime/src/datasets_health_monitor.rs

```diff
@@ -123,6 +127,63 @@ impl DatasetsHealthMonitor {
         Ok(table)
     }
 
+    // returns a list of dataset names that had successful queries against them in the last 10 minutes
+    pub async fn get_recently_accessed_datasets(
+        df_ctx: Arc<SessionContext>,
+    ) -> Result<Arc<HashSet<String>>> {
+        let query = format!(
+            "
+SELECT labels.datasets AS datasets
+FROM runtime.task_history
+WHERE labels.datasets IS NOT NULL
+AND NOW() < end_time + INTERVAL '{DATASET_UNAVAILABLE_THRESHOLD_MINUTES}' MINUTE
+AND labels.tags NOT LIKE '%error%'"
+        );
+        let plan = df_ctx
+            .sql(&query)
+            .await
+            .context(UnableToGetRecentlyAccessedDatasetsSnafu)?;
+        let stream = plan
+            .execute_stream()
+            .await
+            .context(UnableToGetRecentlyAccessedDatasetsSnafu)?;
+        let datasets_with_recent_activity = stream
+            .try_collect::<Vec<RecordBatch>>()
+            .await
+            .context(UnableToGetRecentlyAccessedDatasetsSnafu)?;
+
+        let datasets_with_recent_activity: Vec<String> = datasets_with_recent_activity
+            .iter()
+            .flat_map(|r| match r.column(0).data_type() {
+                arrow::datatypes::DataType::Utf8 => r
+                    .column(0)
+                    .as_string::<i32>()
+                    .into_iter()
+                    .flatten()
+                    .collect::<Vec<_>>(),
+                arrow::datatypes::DataType::LargeUtf8 => r
+                    .column(0)
+                    .as_string::<i64>()
+                    .into_iter()
+                    .flatten()
+                    .collect::<Vec<_>>(),
+                _ => {
+                    unreachable!("Unexpected data type from task_history query result");
+                }
+            })
+            .flat_map(|datasets| datasets.split(',').collect::<Vec<_>>())
+            .map(std::string::ToString::to_string)
+            .collect::<Vec<_>>();
```

Comment:
Is it possible to directly do a `.collect::<HashSet<_>>()` here? so we don't need to iterate the Vec<String> once more to craete the hashset of datasets_with_recent_activity?